index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
Aren't these concerns all problems inherent to js? It wouldn't matter if you removed the `Semigroup` and `Monoid` implementations, you can still bork yourself by passing the wrong arguments. 

If you don't give a function to `map` you're screwed. 

If you don't call `ap` on a `Maybe (a -> b)` with a `Maybe a` you're screwed.

If you don't give bind a function `a -> Maybe b` you're screwed.

There are plenty of ways to shoot yourself in the foot with js. Is there a reason `Semigroup`/`Monoid` should be held to some stronger requirements?
You can try to check arguments and return values all you'd like, there there are still ways to mess that up as well.

That said, if the implementation should change, then go for it. You can change the semantics of what `concat` does as well. You can make it so it short circuits on the first argument.

```javascript
Nothing.prototype.concat = function(m) {
  return m;
};
Just.prototype.concat = function(m) {
  return this;
};
```
This is still lawful, just different semantics.